test/store_unittest.cc

```diff
@@ -0,0 +1,229 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() const {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+  Configuration config;
```

Comment:
Do we actually need a local copy of the config? I'm only seeing it being used in the constructor.

Reply:
Done. Good catch.

Reply:
Actually, we do need a local copy of the config, to make sure it gets properly cleaned up. The store does not own the config reference, so the new code leaks. In the previous version, the `MetadataStoreTest` object owned the `Configuration` object, and cleanup happened automatically. Can we please revert this change?

Reply:
Done.